packages/ibm-products-web-components/src/components/big-number/big-number.ts

```diff
@@ -0,0 +1,198 @@
+/**
+ * @license
+ *
+ * Copyright IBM Corp. 2025
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import { LitElement, html, nothing } from 'lit';
+import { property } from 'lit/decorators.js';
+import { classMap } from 'lit/directives/class-map.js';
+import { carbonElement as customElement } from '@carbon/web-components/es/globals/decorators/carbon-element.js';
+import { iconLoader } from '@carbon/web-components/es/globals/internal/icon-loader.js';
+import { CarbonIcon } from '@carbon/web-components/es/globals/internal/icon-loader-utils';
+import ArrowUp16 from '@carbon/icons/es/arrow--up/16';
+import ArrowUp20 from '@carbon/icons/es/arrow--up/20';
+import ArrowUp24 from '@carbon/icons/es/arrow--up/24';
+
+import { prefix } from '../../globals/settings';
+import { getSupportedLocale } from '../../globals/js/utils/getSupportedLocale';
+import './big-number-skeleton';
+import {
+  BigNumberSize,
+  BigNumberSizeValues,
+  Characters,
+  DefaultLocale,
+  formatValue,
+  getIconSize,
+} from './constants';
+import styles from './big-number.scss?lit';
+
+const blockClass = `${prefix}--big-number`;
+
+/**
+ * @element c4p-big-number
+ * @slot label - Header area that displays the label above the value. This slot can be customized to include an info icon and a tooltip for additional context.
+ * @slot trending-icon – Displays an icon indicating trend direction. Can be customized to show a downward arrow when the trend is not upward.
+ * @slot icon-button – Displays an icon button next to `value`.
+ */
+@customElement(`${prefix}-big-number`)
+class CDSBigNumber extends LitElement {
+  @property({ type: Number })
+  fractionDigits? = 1;
+
+  @property({ type: String })
+  label: string | undefined;
+
+  @property({ type: Boolean })
+  loading? = false;
+
+  @property({ type: String })
+  locale? = DefaultLocale;
+
+  @property({ type: Boolean })
+  percentage? = false;
+
+  @property({ type: String })
+  size?: BigNumberSizeValues = BigNumberSize.Default;
+
+  @property({ type: Number })
+  total?: number;
+
+  @property({ type: Boolean })
+  trending? = false;
+
+  @property({ type: Boolean })
+  truncate? = true;
+
+  @property({ type: Number })
+  value?: number;
+
+  private _getSupportedLocale(locale: string): string {
+    return getSupportedLocale(locale, DefaultLocale);
+  }
+
+  private _formatValue(
+    supportedLocale: string,
+    value: number,
+    fractionDigits: number,
+    truncate: boolean
+  ): string {
+    return formatValue(supportedLocale, value, fractionDigits, truncate);
+  }
+
+  private _getTruncatedValue(): string {
```

Comment:
this function and the one underneath it are basically identical, so we're replicating code. we should find a way to consolidate here.